### PR TITLE
Add test_apply_errata_using_default_content_view

### PR DESCRIPTION
**Pull request by swadeley**
Hello

This adds a test for an errata in Library, using Default Content View .
This test relates to the change to Pulp3.

Depends on: [ Update to RHBA-2021:1314 for rhst7 #8617 ](https://github.com/SatelliteQE/robottelo/pull/8617)

### CODE REVIEW

**Comment by tpapaioa** [comment]
A couple code check errors:

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

tests/foreman/cli/test_errata.py:1335:100: E501 line too long (101 > 99 characters)
tests/foreman/cli/test_errata.py:1337:100: E501 line too long (125 > 99 characters)
```

**Comment by swadeley** [comment]
@tpapaioa fixed that, thank you

**Review by mshriver — APPROVED** [review]

File: tests/foreman/cli/test_errata.py [hunk 1]
```diff
@@ -1275,3 +1280,114 @@ def test_positive_check_errata_dates(module_org):
...
+    # note time for later wait_for_tasks include 2 mins margin of safety.
+    timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
+    # Update errata from Library, i.e. Default CV
+    chost.run(f'yum -y update --advisory {REAL_0_ERRATA_ID}')
+    # Wait for upload profile event (in case Satellite system slow)
+    host = entities.Host().search(query={'search': f'name={chost.hostname}'})
+    wait_for_tasks(
+        search_query=(
+            'label = Actions::Katello::Host::UploadProfiles'
+            f' and resource_id = {host[0].id}'
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1385]:
```suggestion
    wait_for_tasks(
        search_query=(
            'label = Actions::Katello::Host::UploadProfiles'
            f' and resource_id = {chost.nailgun_host.id}'
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_errata.py [hunk 2]
```diff
@@ -1275,3 +1280,114 @@ def test_positive_check_errata_dates(module_org):
...
+   The applicability tests using Default Content View are related to the introduction of Pulp3.
+   """
+
+
+@pytest.fixture(scope='module')
+def default_contentview(module_manifest_org):
+    return entities.ContentView(organization=module_manifest_org, name=DEFAULT_CV).search()
+
+
+@pytest.fixture(scope="module")
```
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1313]: ok

File: tests/foreman/cli/test_errata.py [hunk 3]
```diff
@@ -1275,3 +1280,114 @@ def test_positive_check_errata_dates(module_org):
...
+    rhel77_contenthost_module.run(
+        'rpm -Uvh https://download-ib01.fedoraproject.org/pub/epel/7/'
+        'x86_64/Packages/p/python2-psutil-5.6.7-1.el7.x86_64.rpm'
+    )
+    rhel77_contenthost_module.install_katello_ca()
+    rhel77_contenthost_module.register_contenthost(module_manifest_org.label, new_module_ak.name)
+    host = entities.Host().search(query={'search': f'name={rhel77_contenthost_module.hostname}'})
+    host_id = host[0].id
+    host_content = entities.Host(id=host_id).read_json()
+    assert host_content["subscription_status"] == 0
```
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1350]: ok

File: tests/foreman/cli/test_errata.py [hunk 4]
```diff
@@ -1275,3 +1280,114 @@ def test_positive_check_errata_dates(module_org):
...
+    """A RHEL77 Content Host that has applicable errata and registered to Library"""
+    # python-psutil is obsoleted by python2-psutil, so install older python2-psutil for errata test
+    rhel77_contenthost_module.run(
+        'rpm -Uvh https://download-ib01.fedoraproject.org/pub/epel/7/'
+        'x86_64/Packages/p/python2-psutil-5.6.7-1.el7.x86_64.rpm'
+    )
+    rhel77_contenthost_module.install_katello_ca()
+    rhel77_contenthost_module.register_contenthost(module_manifest_org.label, new_module_ak.name)
+    host = entities.Host().search(query={'search': f'name={rhel77_contenthost_module.hostname}'})
+    host_id = host[0].id
```
[the remark this answers is not in the document]
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: done

File: tests/foreman/cli/test_errata.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: done

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/cli/test_errata.py -k test_apply_errata_using_default_content_view
=============================== test session starts ========================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo_2/robottelo, configfile: pyproject.toml
plugins: mock-3.6.1, forked-1.3.0, services-2.2.1, cov-2.12.0, ibutsu-1.14.1, reportportal-5.0.8, xdist-2.2.1
collected 38 items / 37 deselected / 1 selected                                                                                                                                                   

tests/foreman/cli/test_errata.py .                                                                                                                                                          [100%]

========================= 1 passed, 37 deselected, 1 warning in 419.64s (0:06:59) ===========================

File: tests/foreman/cli/test_errata.py [hunk 5]
```diff
@@ -1275,3 +1280,112 @@ def test_positive_check_errata_dates(module_org):
...
+    # python-psutil is obsoleted by python2-psutil, so install older python2-psutil for errata test
+    rhel77_contenthost_module.run(
+        'rpm -Uvh https://download-ib01.fedoraproject.org/pub/epel/7/'
+        'x86_64/Packages/p/python2-psutil-5.6.7-1.el7.x86_64.rpm'
+    )
+    rhel77_contenthost_module.install_katello_ca()
+    rhel77_contenthost_module.register_contenthost(module_manifest_org.label, new_module_ak.name)
+    host_id = rhel77_contenthost_module.nailgun_host.id
+    host_content = entities.Host(id=host_id).read_json()
+    assert host_content['subscription_status'] == 0
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1349]: Is this not already an attribute of the nailgun_host object?

Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1349]: @JacobCallahan I don't see it:
>>> result = rhel77_contenthost_module.nailgun_host
>>> result
nailgun.entities.Host(architecture=nailgun.entities.Architecture(id=1), build=False, capabilities=['build'], comment=None, compute_profile=None,     
compute_resource=None, content_facet_attributes={'id': 2, 'uuid': '165d5c6d-0bbb-49c1-9609-8c70161ef80e', 'content_view_id': 4, 'content_view_name': 
'Default Organization View', 'lifecycle_environment_id': 4, 'lifecycle_environment_name': 'Library', 'content_source_id': None,                      
'content_source_name': None, 'kickstart_repository_id': None, 'kickstart_repository_name': None, 'errata_counts': {'security': 0, 'bugfix': 0,       
'enhancement': 0, 'total': 0}, 'applicable_package_count': 0, 'upgradable_package_count': 0, 'applicable_module_stream_count': 0,                    
'upgradable_module_stream_count': 0, 'content_view': {'id': 4, 'name': 'Default Organization View'}, 'lifecycle_environment': {'id': 4, 'name':      
'Library'}, 'content_source': None, 'kickstart_repository': None}, domain=nailgun.entities.Domain(id=1), enabled=True, environment=None,             
hostgroup=None, image=None, ip='10.1.3.225', location=nailgun.entities.Location(id=2), mac='56:6f:0d:72:00:2e', managed=False, medium=None,          
model=nailgun.entities.Model(id=1), name='dhcp-3-225.vms.sat.rdu2.redhat.com', operatingsystem=nailgun.entities.OperatingSystem(id=2),               
organization=nailgun.entities.Organization(id=5), owner=nailgun.entities.User(id=1), provision_method='build', ptable=None, puppet_ca_proxy=None,    
puppet_proxy=None, realm=None, subnet=None, uuid=None, id=3, owner_type='User')

File: tests/foreman/cli/test_errata.py [hunk 6]
```diff
@@ -1275,3 +1280,112 @@ def test_positive_check_errata_dates(module_org):
...
+        environment=entities.LifecycleEnvironment(id=module_manifest_org.library.id),
+        organization=module_manifest_org,
+    ).create()
+    # Ensure tools repo is enabled in the activation key
+    new_module_ak.content_override(
+        data={'content_overrides': [{'content_label': REPOS['rhst7']['id'], 'value': '1'}]}
+    )
+    # Fetch available subscriptions
+    subs = entities.Subscription(organization=module_manifest_org).search()
+    assert len(subs) > 0
```
Comment (mshriver) [inline tests/foreman/cli/test_errata.py:1326]: `> 0` is redundant here.

File: tests/foreman/cli/test_errata.py [hunk 7]
```diff
@@ -1275,3 +1280,112 @@ def test_positive_check_errata_dates(module_org):
...
+    # Add default subscription to activation key
+    sub_found = False
+    for sub in subs:
+        if sub.name == DEFAULT_SUBSCRIPTION_NAME:
+            new_module_ak.add_subscriptions(data={'subscription_id': sub.id})
+            sub_found = True
```
Comment (mshriver) [inline tests/foreman/cli/test_errata.py:1332]: Non blocking, but this block seems a bit clunky, starting with getting the subscriptions.

The only subscription you need here is the default subscription, and then this block exists to find that, and then add it to the ak.

Consider a pattern more like the following, which is much more succinct and I believe achieves the same things as L1325-1334

```
subs = entities.Subscription(organization=module_manifest_org).search(
    query={'search': f'name={DEFAULT_SUBSCRIPTION_NAME}'}
)
assert subs
new_module_ak.add_subscriptions(data={'subscription_id': subs[0].id})

return new_module_ak

```

File: tests/foreman/cli/test_errata.py (hunk 5 above)
Comment (mshriver) [reply tests/foreman/cli/test_errata.py:1349]: Agreed, creation of `host_content` from `entities.Host` is redundant here. `rhel77_contenthost_module.nailgun_host` is the same entity as `entities.Host(id=host_id)`.

You should at the least be able to just directly call `read_json()` on `rhel77_contenthost_module`

Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1349]: OK, this works:
`In [8]: rhel77_contenthost_module.nailgun_host.read_json()['subscription_status']`
`Out[8]: 0`

File: tests/foreman/cli/test_errata.py (hunk 7 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1332]: Hello, that code works but only when used without `name=`, for some reason search is only working with `name=` if the search string is double quoted. I think its safe to leave it out.

Comment (mshriver) [reply tests/foreman/cli/test_errata.py:1332]: ack, yeah full disclose that was typed without running it - was focusing more on the pattern at hand then the specific behavior of search.  Sounds a bit odd though, but if you can reliably filter that instance in the search call, I would.

Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: OK, I have made that change, thank you

**Review by mshriver — APPROVED** [review]
Looks like all comments from reviewers have been addressed, nice work!

**Comment by swadeley** [comment]
Thank you

**Pull request merged**